tasks/deployers.ts

```diff
@@ -0,0 +1,159 @@
+): Promise<
+  Pick<
+    ContractAddresses,
+    | 'CustomSmartWallet'
+    | 'CustomSmartWalletFactory'
+    | 'CustomSmartWalletDeployVerifier'
+    | 'CustomSmartWalletRelayVerifier'
+  >
+> => {
+  const customSmartWalletF = await ethers.getContractFactory(
+    'CustomSmartWallet'
+  );
+  const { address: customSmartWalletAddress } =
+    await customSmartWalletF.deploy();
+  const customSmartWalletFactoryF = await ethers.getContractFactory(
+    'CustomSmartWalletFactory'
+  );
+  const { address: customSmartWalletFactoryAddress } =
+    await customSmartWalletFactoryF.deploy(customSmartWalletAddress);
+  const customSmartWalletDeployVerifierF = await ethers.getContractFactory(
+    'CustomSmartWalletDeployVerifier'
+  );
+  const { address: customSmartWalletDeployVerifierAddress } =
+    await customSmartWalletDeployVerifierF.deploy(
+      customSmartWalletFactoryAddress
+    );
+  const relayVerifierF = await ethers.getContractFactory('RelayVerifier');
+  const { address: customSmartWalletRelayVerifierAddress } =
+    await relayVerifierF.deploy(customSmartWalletFactoryAddress);
+
+  return {
+    CustomSmartWalletFactory: customSmartWalletFactoryAddress,
+    CustomSmartWallet: customSmartWalletAddress,
+    CustomSmartWalletDeployVerifier: customSmartWalletDeployVerifierAddress,
+    CustomSmartWalletRelayVerifier: customSmartWalletRelayVerifierAddress,
+  };
+};
+
+export const deployDefaultSmartWallet = async (
+  ethers: HardhatEthersHelpers
+): Promise<
+  Pick<
+    ContractAddresses,
+    'SmartWallet' | 'SmartWalletFactory' | 'RelayVerifier' | 'DeployVerifier'
+  >
+> => {
+  const smartWalletF = await ethers.getContractFactory('SmartWallet');
+  const { address: smartWalletAddress } = await smartWalletF.deploy();
+
+  const smartWalletFactoryF = await ethers.getContractFactory(
+    'SmartWalletFactory'
+  );
+  const { address: smartWalletFactoryAddress } =
+    await smartWalletFactoryF.deploy(smartWalletAddress);
+
+  const deployVerifierF = await ethers.getContractFactory('DeployVerifier');
+  const { address: deployVerifierAddress } = await deployVerifierF.deploy(
+    smartWalletFactoryAddress
+  );
+
+  const relayVerifierF = await ethers.getContractFactory('RelayVerifier');
+  const { address: relayVerifierAddress } = await relayVerifierF.deploy(
+    smartWalletFactoryAddress
+  );
+
+  return {
+    SmartWallet: smartWalletAddress,
+    SmartWalletFactory: smartWalletFactoryAddress,
+    DeployVerifier: deployVerifierAddress,
+    RelayVerifier: relayVerifierAddress,
+  };
+};
+export const deployRelayHub = async (
+  ethers: HardhatEthersHelpers
+): Promise<Pick<ContractAddresses, 'RelayHub' | 'Penalizer'>> => {
+  const relayHubF = await ethers.getContractFactory('RelayHub');
+  const penalizerF = await ethers.getContractFactory('Penalizer');
+  const { address: penalizerAddress } = await penalizerF.deploy();
+  const { address: relayHubAddress } = await relayHubF.deploy(
+    penalizerAddress,
+    1,
```

Comment:
I never think about this before, but what about if we want to change the parameters for the relayHub, should we hardcode it?

Reply:
Yes, I've realised this while doing this change. It's always been hard-coded and I don't like it either. I think in future we could provide an additional script just for deploying the relayHub, that will receive all the parameters required instead of hard-coding it 🤦🏽‍♂️